opnode/rollup/derive/payload_attributes.go

```diff
@@ -206,7 +206,7 @@ func UserDeposits(receipts []*types.Receipt, depositContractAddr common.Address)
 			continue
 		}
 		for _, log := range rec.Logs {
-			if log.Address == depositContractAddr {
+			if log.Address == depositContractAddr && log.Topics[0] == DepositEventABIHash {
```

Comment:
@trianglesphere: I added this check since the `OptimismPortal` contract can emit multiple different event types.

Reply:
You also need a check for `len(log.Topics) > 0` here. We do check that the ABI is correct in the `UnmarshalLogEvent`, but halt instead of ignoring the error thus adding the check here is important.